app/pulp/app/response.py

```diff
@@ -0,0 +1,32 @@
+from gettext import gettext as _
+
+from rest_framework.response import Response
+
+
+class OperationPostponedResponse(Response):
+    """
+    An HTTP response class for returning 202 and a list of spawned tasks.
+
+    This reponse object should be used by views that dispatch asynchronous tasks. Most common use
+    case is for sync and publish operations. When the task_group_id is specified, the task_results
+    list should only contain 10 items. Task groups can consist of many tasks, so returning their
+    ids should occur in a paginated fashion.
+    """
+
+    def __init__(self, task_results, task_group_id=None):
+        """
+        :param task_results: List of AsyncResult objects that are used to generate the response.
+        :type task_group_id: List of AsyncResult
+        :param task_group_id: Task group id, if any, for all the task results
+        :type task_group_id: str
+        """
+        if not isinstance(task_results, list):
+            raise TypeError(_('task_results must be an instance of list.'))
```

Comment:
If that's the only reason, then you may not need this check at all. The ``for`` loop will raise a ``TypeError`` if the object isn't iterable.

... unless of course an AsyncResult is iterable. Crazier things have happened. :)